pkg/asset/machines/baremetal/hosts.go

```diff
@@ -116,10 +116,31 @@ func Hosts(config *types.InstallConfig, machines []machineapi.Machine, userDataS
 		return nil, fmt.Errorf("no baremetal platform in configuration")
 	}
 
+	numRequiredArbiters := replicaCount(config.Arbiter)
+	numArbiters := 0
+
 	numRequiredMasters := len(machines)
 	numMasters := 0
 	for _, host := range config.Platform.BareMetal.Hosts {
 
+		// We only infer arbiter hosts if we are in an arbiter deployment
+		if config.IsArbiterEnabled() {
+			// If we know the host is an arbiter and the role isn't an empty value (i.e. explicitly defined by user)
+			// no further investigating is needed, skip creating resources
+			if host.IsArbiter() && host.Role != "" {
```

Comment:
nit: the `&& host.Role != ""` part is redundant here (since `IsArbiter()` returns `host.Role == "arbiter"`)

Reply:
Ah, yup, removing.